storage/innobase/fil/fil0crypt.cc

```diff
@@ -232,40 +233,58 @@ fil_space_crypt_t::fil_space_crypt_t(uint new_type, uint new_min_key_version,
       rotate_state(),
       encryption_rotation(encryption_rotation),
       tablespace_key(NULL) {
+  mutex_create(LATCH_ID_FIL_CRYPT_START_ROTATE_MUTEX, &start_rotate_mutex);
+  mutex_create(LATCH_ID_FIL_CRYPT_DATA_MUTEX, &mutex);
+
   key_id = new_key_id;
   if (my_random_bytes(iv, sizeof(iv)) != MY_AES_OK)  // TODO:Robert: This can
                                                      // return error and because
                                                      // of that it should not be
                                                      // in constructor
     type = 0;  // TODO:Robert: This is temporary to get rid of unused variable
                // problem
-  mutex_create(LATCH_ID_FIL_CRYPT_START_ROTATE_MUTEX, &start_rotate_mutex);
-  mutex_create(LATCH_ID_FIL_CRYPT_DATA_MUTEX, &mutex);
-  // locker = crypt_data_scheme_locker; // TODO:Robert: Co to za locker, nie
-  // mogę znaleść jego definicji nawet w mariadb
-  type = new_type;
+  if (new_uuid != nullptr && strlen(new_uuid) > 0) {
+    memcpy(uuid, new_uuid, ENCRYPTION_SERVER_UUID_LEN);
+    uuid[ENCRYPTION_SERVER_UUID_LEN] = '\0';
+  } else {
+    uuid[0] = '\0';
+  }
```

Comment:
Encryption::s_uuid is not always equal to server_uuid. It is a bit of weird thing. When a table encrypted with Master Key is read and this table was copied over from some different location - for instance backup, and its Master Key Id (which does not include uuid) is higher than currently discovered Id - the Encryption::s_uuid will be replaced with UUID read from this table header. This can be seen in Encryption::decode_encryption_information. It is better to keep keyring encryption away from it.

Reply:
Ok. I am thinking if we can avoid uuid in every fil_space_t/fil_crypt_t and simplify things.. (but also wondering how it works with MK.. if it only has one "global" uuid).